ank/src/cli_commands.rs

```diff
@@ -223,42 +224,53 @@ impl CliCommands {
         let _ = self.task.await;
     }
 
+    async fn get_complete_state(&mut self, object_field_mask: &Vec<String>) -> Result<Box<CompleteState>, CliError> {
+        output_debug!("get_complete_state: object_field_mask={:?} ", object_field_mask);
+
+        // send complete state request to server
+        self.to_server
+            .request_complete_state(RequestCompleteState {
+                request_id: self.cli_name.to_owned(),
+                field_mask: object_field_mask.clone(),
+            })
+            .await.map_err(|err| CliError::ExecutionError(err.to_string()))?;
+
+        let poll_complete_state_response = async {
+            loop {
+                match self.from_server.recv().await {
+                    Some(ExecutionCommand::CompleteState(res)) => return res,
+                    Some(_) | None => (),
```

Comment:
I the `None` case we can actually directly exit, as the channel was closed from the other side:
```suggestion
                    None => return Err("Channel preliminary closed."),
                    _ => (),
```

But this requires also handling the return values differently (see below) ...